tools/generate_app.dart

```diff
@@ -0,0 +1,133 @@
+import 'dart:convert';
+import 'dart:io';
+
+/// Phosphor Icons generator
+/// reads the phosphor json and generates a dart class
+/// with all the phosphor icons constants
+void main(List<String> arguments) {
+  final jsonFile = File(arguments.first);
+  assert(!jsonFile.existsSync());
+
+  final icons = json.decode(jsonFile.readAsStringSync())['icons'];
+
+  final fileLines = [
+    """// Auto Generated File
+
+import 'package:phosphor_flutter/phosphor_flutter.dart';
+
+abstract class AllIcons {""",
+  ];
+  final mapGetterContent = <String>[
+    '''
+  static Map<String, PhosphorIconData> get allIconsAsMap => {
+      ...boldIcons,
+      ...regularIcons,
+      ...thinIcons,
+      ...lightIcons,
+      ...fillIcons,
+    };''',
+  ];
+
+  final mapBoldGetterContent = <String>[
+    '\n  static Map<String, PhosphorIconData> get boldIcons => {',
+    '};'
+  ];
+
+  final mapRegularGetterContent = <String>[
+    '\n  static Map<String, PhosphorIconData> get regularIcons => {',
+    '};'
+  ];
+
+  final mapThinGetterContent = <String>[
+    '\n  static Map<String, PhosphorIconData> get thinIcons => {',
+    '};'
+  ];
+
+  final mapLightGetterContent = <String>[
+    '\n  static Map<String, PhosphorIconData> get lightIcons => {',
+    '};'
+  ];
+
+  final mapFillGetterContent = <String>[
+    '\n  static Map<String, PhosphorIconData> get fillIcons => {',
+    '};'
+  ];
+
+  final iconsGetterContent =
+      '\n  static List<PhosphorIconData> get icons => allIconsAsMap.values.toList();';
+  final namesGetterContent =
+      '\n  static List<String> get names => allIconsAsMap.keys.toList();';
+
+  icons.forEach((icon) {
```

Comment:
Would be nice to add fail-fast behavior here -- if all 5 weights are not present, codegen should fail catastrophically.

Reply:
thanks to this I could see a bug in the generator. I fixed it and now works ans checks if an icon don't have the five styles